mlfinlab/ensemble/sb_bagging.py

```diff
@@ -0,0 +1,660 @@
+        If None, the random number generator is the RandomState instance used
+        by `np.random`.
+    verbose : int, optional (default=0)
+        Controls the verbosity when fitting and predicting.
+    Attributes
+    ----------
+    base_estimator_ : estimator
+        The base estimator from which the ensemble is grown.
+    estimators_ : list of estimators
+        The collection of fitted base estimators.
+    estimators_samples_ : list of arrays
+        The subset of drawn samples (i.e., the in-bag samples) for each base
+        estimator. Each subset is defined by an array of the indices selected.
+    estimators_features_ : list of arrays
+        The subset of drawn features for each base estimator.
+    classes_ : array of shape = [n_classes]
+        The classes labels.
+    n_classes_ : int or list
+        The number of classes.
+    oob_score_ : float
+        Score of the training dataset obtained using an out-of-bag estimate.
+    oob_decision_function_ : array of shape = [n_samples, n_classes]
+        Decision function computed with out-of-bag estimate on the training
+        set. If n_estimators is small it might be possible that a data point
+        was never left out during the bootstrap. In this case,
+        `oob_decision_function_` might contain NaN.
+    References
+    ----------
+    .. [1] L. Breiman, "Pasting small votes for classification in large
+           databases and on-line", Machine Learning, 36(1), 85-103, 1999.
+    .. [2] L. Breiman, "Bagging predictors", Machine Learning, 24(2), 123-140,
+           1996.
+    .. [3] T. Ho, "The random subspace method for constructing decision
+           forests", Pattern Analysis and Machine Intelligence, 20(8), 832-844,
+           1998.
+    .. [4] G. Louppe and P. Geurts, "Ensembles on Random Patches", Machine
+           Learning and Knowledge Discovery in Databases, 346-361, 2012.
+    """
+
+    def __init__(self,
+                 triple_barrier_events,
+                 price_bars,
+                 base_estimator=None,
+                 n_estimators=10,
+                 max_samples=1.0,
+                 max_features=1.0,
+                 bootstrap_features=False,
+                 oob_score=False,
+                 warm_start=False,
+                 n_jobs=None,
+                 random_state=None,
+                 verbose=0,
+                 ):
+        super().__init__(
+            triple_barrier_events=triple_barrier_events,
+            price_bars=price_bars,
+            base_estimator=base_estimator,
+            n_estimators=n_estimators,
+            max_samples=max_samples,
+            max_features=max_features,
+            bootstrap_features=bootstrap_features,
+            oob_score=oob_score,
+            warm_start=warm_start,
+            n_jobs=n_jobs,
+            random_state=random_state,
+            verbose=verbose)
+
+    def _validate_estimator(self):
```

Comment:
Do you need this function if the parent class already has it?

Reply:
Yeah, because you need to specify what is default base_estimator (I remember it didn't work until I declared this function)